frontend/src/pages/MilestonesEditor.tsx

```diff
@@ -86,8 +87,13 @@ export const MilestonesEditor = () => {
       SortingOrders.DESCENDING,
     )(Array.from(ratedTasks.values()));
 
+    if (areInitialVersions) {
+      setAreInitialVersions(false);
+      if (!newVersionLists[ROADMAP_LIST_ID].length) setExpandUnordered(false);
+    }
+
```

Comment:
It's better to separate this into its own useEffect, otherwise this useEffect would run once unnecessarily (when areInitialVersions is set to false)

It would look something like this

`useEffect(() => {
    if (!areInitialVersions || !versionLists[ROADMAP_LIST_ID]) return;
    setAreInitialVersions(false);
    if (!versionLists[ROADMAP_LIST_ID].length) setExpandUnordered(false);
  }, [areInitialVersions, versionLists]);`